### PR TITLE
feat(buy/sell): decouple enter amount sell components from buy

### PR DESCRIPTION
**Pull request by dkremniov-bc**
## Description
In order to be able to make quote/order changes to buy flow in the future PRs, we need to decouple buy and sell enter amount components. We also need to make sure that all the redirects to enter amount step are pointing to either buy or sell based on the context.

## How to review this PR
Please ignore newly added files under `SellEnterAmount` directory - they are just copies of existing `EnterAmount`. There is also a weird merge in the commit history in order to preserve the history of duplicated files.

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/modals/BuySell/CryptoSelection/CryptoSelector/index.tsx [hunk 1]
```diff
@@ -134,7 +134,7 @@ class CryptoSelector extends React.Component<InjectedFormProps<{}, Props> & Prop
       fiatCurrency: this.props.walletCurrency,
       orderType: this.state.orderType,
       pair,
-      step: 'ENTER_AMOUNT',
+      step: 'SELL_ENTER_AMOUNT',
```
Comment (blockdylanb) [inline packages/blockchain-wallet-v4-frontend/src/modals/BuySell/CryptoSelection/CryptoSelector/index.tsx:137]: Should be using `getEnterAmountStepType` here?

File: packages/blockchain-wallet-v4-frontend/src/modals/BuySell/SellEnterAmount/Checkout/Payment/Payment.styles.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import styled from 'styled-components'
+
+export const DisplayIconAligned = styled.div`
```
Comment (blockdylanb) [inline packages/blockchain-wallet-v4-frontend/src/modals/BuySell/SellEnterAmount/Checkout/Payment/Payment.styles.ts:3]: There's a `<Flex />` component in constellation that could replace this.

File: packages/blockchain-wallet-v4-frontend/src/modals/BuySell/SellEnterAmount/Checkout/BaseQuote/index.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  margin-left: 0.5rem;
+  width: 100%;
+  font-size: 1rem;
+  font-weight: 500;
+  color: ${(props) => props.theme.grey800};
+`
+
+const IconBackground = styled.div`
+  display: flex;
+  justify-content: center;
```
[the remark this answers is not in the document]
Comment (JeremyMMA0444) [reply packages/blockchain-wallet-v4-frontend/src/modals/BuySell/SellEnterAmount/Checkout/BaseQuote/index.tsx:30]: Pax

[the remark this answers is not in the document]
Comment (JeremyMMA0444) [reply packages/blockchain-wallet-v4-frontend/src/modals/BuySell/SellEnterAmount/Checkout/BaseQuote/index.tsx:30]: You-Van-help-ne-bro

[the remark this answers is not in the document]
Comment (JeremyMMA0444) [reply packages/blockchain-wallet-v4-frontend/src/modals/BuySell/SellEnterAmount/Checkout/BaseQuote/index.tsx:30]: You-Can-help-ne

[the remark this answers is not in the document]
Comment (JeremyMMA0444) [reply packages/blockchain-wallet-v4-frontend/src/modals/BuySell/SellEnterAmount/Checkout/BaseQuote/index.tsx:30]: You-can-Help-me

**Review by JeremyMMA0444 — APPROVED** [review]

**Comment by JeremyMMA0444** [comment]
@blockdylanb Can-help-me-=please=

**Review by blockdylanb — DISMISSED** [review]

File: packages/blockchain-wallet-v4-frontend/src/data/components/buySell/utils.ts [hunk 4]
```diff
@@ -39,3 +39,11 @@ export const getQuoteRefreshConfig = ({
     totalMs: millisecondsUntilRefresh
   }
 }
+
+export const getEnterAmountStepType = (orderType?: BSOrderActionType) => {
+  if (!orderType || orderType === OrderType.BUY) {
+    return 'ENTER_AMOUNT'
+  }
+
+  return 'SELL_ENTER_AMOUNT'
```
[the remark this answers is not in the document]
Comment (dkremniov-bc) [reply packages/blockchain-wallet-v4-frontend/src/data/components/buySell/utils.ts:48]: Yes, it is something that I am looking to do with a large refactoring; however, right now it is not feasible.

File: packages/blockchain-wallet-v4-frontend/src/modals/BuySell/SellEnterAmount/Checkout/ActionButton/index.tsx [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+          disabled={disabled || disableInsufficientEth || dailyLimitExceeded}
+        >
+          {props.submitting ? (
+            <HeartbeatLoader height='16px' width='16px' color='white' />
+          ) : (
+            <FormattedMessage
+              id='buttons.preview_buysell'
+              defaultMessage='Preview {orderType}'
+              values={{
+                orderType: props.orderType === OrderType.BUY ? 'Buy' : 'Sell'
```
[the remark this answers is not in the document]
Comment (dkremniov-bc) [reply packages/blockchain-wallet-v4-frontend/src/modals/BuySell/SellEnterAmount/Checkout/ActionButton/index.tsx:105]: As I mentioned, files in `SellEnterAmount` are just copies of existing `EnterAmount`. However, that is a good point and I will make a note for myself when I will further refactoring these components.

File: packages/blockchain-wallet-v4-frontend/src/modals/BuySell/SellEnterAmount/Checkout/index.tsx [hunk 6]
```diff
@@ -0,0 +1,305 @@
...
+    return null
+  }
+
+  if (isLoading) {
+    return <Loading />
+  }
+
+  if (!data) {
+    return null
+  }
```
[the remark this answers is not in the document]
Comment (dkremniov-bc) [reply packages/blockchain-wallet-v4-frontend/src/modals/BuySell/SellEnterAmount/Checkout/index.tsx:254]: Same, just a copy of existing file, but good point.

File: packages/blockchain-wallet-v4-frontend/src/modals/BuySell/CryptoSelection/CryptoSelector/index.tsx (hunk 1 above)
Comment (dkremniov-bc) [reply packages/blockchain-wallet-v4-frontend/src/modals/BuySell/CryptoSelection/CryptoSelector/index.tsx:137]: This is a part of `handleSell` method. We also have `handleBuy` method in this component. I assume `handleSell` will never be used for buy 🤞

File: packages/blockchain-wallet-v4-frontend/src/modals/BuySell/SellEnterAmount/Checkout/Payment/Payment.styles.ts (hunk 2 above)
Comment (dkremniov-bc) [reply packages/blockchain-wallet-v4-frontend/src/modals/BuySell/SellEnterAmount/Checkout/Payment/Payment.styles.ts:3]: This file is just a copy of what we already had. Definitely something to refactor 👍

**Review by dylanbathurst — APPROVED** [review]

**Comment by dkremniov-bc** [comment]
Rebased on latest `development` containing a fix for default payment method loading. Should affect only copied files.

**Pull request merged**